pkg/workflows/wasm/host/standard_test.go

```diff
@@ -0,0 +1,469 @@
+		lt100Exec := NewMockExecutionHelper(t)
+		lt100Exec.EXPECT().GetWorkflowExecutionID().Return(anyId)
+
+		lt100Exec.EXPECT().CallCapability(mock.Anything, mock.Anything).RunAndReturn(setupNodeCallAndConsensusCall(t, 99))
+		lt100Exec.EXPECT().EmitUserLog(mock.Anything).RunAndReturn(func(s string) error {
+			_, err = strconv.ParseUint(s, 10, 64)
+			require.NoError(t, err)
+			return nil
+		}).Once()
+
+		execution2Result, err := m.Execute(t.Context(), anyRequest, lt100Exec)
+		require.NoError(t, err)
+		wrappedValue2, err := values.FromProto(execution2Result.GetValue())
+		require.NoError(t, err)
+		value2, err := wrappedValue2.Unwrap()
+		require.NoError(t, err)
+		require.Equal(t, value1, value2, "Expected the same random number to be generated for the same trigger")
+	})
+
+	t.Run("Different execution id give different randoms", func(t *testing.T) {
+		require.NoError(t, err)
+
+		gte100Exec2 := NewMockExecutionHelper(t)
+		gte100Exec2.EXPECT().GetWorkflowExecutionID().Return("differentId")
+
+		gte100Exec2.EXPECT().CallCapability(mock.Anything, mock.Anything).RunAndReturn(setupNodeCallAndConsensusCall(t, 120))
+
+		executionResult2, err := m.Execute(t.Context(), anyRequest, gte100Exec2)
+		require.NoError(t, err)
+		wrappedValue2, err := values.FromProto(executionResult2.GetValue())
+		require.NoError(t, err)
+		value2, err := wrappedValue2.Unwrap()
+		require.NoError(t, err)
+		require.NotEqual(t, value1, value2, "Expected different random numbers for different triggers")
+	})
+}
+
+func TestSecrets(t *testing.T) {
+	t.Parallel()
+
+	m := makeTestModule(t)
+
+	t.Run("returns the secret value", func(t *testing.T) {
+		result := runSecretTest(t, m, &sdkpb.SecretResponse{
+			Response: &sdkpb.SecretResponse_Secret{
+				Secret: &sdkpb.Secret{
+					Value: "Bar",
+				},
+			},
+		})
+		require.Equal(t, "Bar", result.GetValue().GetStringValue())
+	})
+
+	t.Run("returns an error if the secret doesn't exist", func(t *testing.T) {
+		resp := runSecretTest(t, m, &sdkpb.SecretResponse{
+			Response: &sdkpb.SecretResponse_Error{
+				Error: "could not find secret",
+			},
+		})
+		assert.ErrorContains(t, errors.New(resp.GetError()), "could not find secret")
+	})
+}
+
+func triggerExecuteRequest(t *testing.T, id uint64, trigger proto.Message) *pb.ExecuteRequest {
+	wrappedTrigger, err := anypb.New(trigger)
+	require.NoError(t, err)
+	return &pb.ExecuteRequest{
+		Config: anyTestConfig,
+		Request: &pb.ExecuteRequest_Trigger{
+			Trigger: &sdkpb.Trigger{Id: id, Payload: wrappedTrigger},
+		},
+		MaxResponseSize: uint64(defaultMaxResponseSizeBytes),
+	}
+}
+
+func runWithBasicTrigger(t *testing.T, executor ExecutionHelper) *pb.ExecutionResult {
+	trigger := &basictrigger.Outputs{CoolOutput: anyTestTriggerValue}
+	executeRequest := triggerExecuteRequest(t, 0, trigger)
+	m := makeTestModule(t)
+	response, err := m.Execute(t.Context(), executeRequest, executor)
+	require.NoError(t, err)
+	return response
+}
+
+func makeTestModule(t *testing.T) *module {
```

Comment:
@nolag Can you document how this works in the README? I was very confused and had to hunt around the code to figure out what was happening